mrp_unbuild_valuation_layer_link/tests/test_mrp__unbuild_valuation_layer_link.py

```diff
@@ -0,0 +1,56 @@
+from odoo.tests import Form
+
+from odoo.addons.mrp.tests.common import TestMrpCommon
+
+
+class TestUnbuild(TestMrpCommon):
+    def setUp(self):
+        super(TestUnbuild, self).setUp()
+        self.stock_location = self.env.ref("stock.stock_location_stock")
+        self.env.ref("base.group_user").write(
+            {"implied_ids": [(4, self.env.ref("stock.group_production_lot").id)]}
+        )
+
+    def test_unbuild_with_valuation_layer(self):
+        """This test creates an Unbuild order from a Manufacturing order and then check if the
+        valuation button links to the valuation layer of the order.
+        """
+        mo, bom, p_final, p1, p2 = self.generate_mo()
+
+        self.env["stock.quant"]._update_available_quantity(p1, self.stock_location, 100)
+        self.env["stock.quant"]._update_available_quantity(p2, self.stock_location, 5)
+        mo.action_assign()
+
+        mo_form = Form(mo)
+        mo_form.qty_producing = 5.0
+        mo = mo_form.save()
+        mo.button_mark_done()
+
+        x = Form(self.env["mrp.unbuild"])
+        x.product_id = p_final
+        x.bom_id = bom
+        x.product_qty = 5
+        unbuild = x.save()
+        unbuild.action_unbuild()
+
+        result = unbuild.action_view_stock_valuation_layers()
+        self.assertEqual(
+            result["res_model"],
+            "stock.valuation.layer",
+            "You should access to the model stock.valuation.layer",
+        )
+        domain = [
+            (
+                "id",
+                "in",
+                (
+                    unbuild.produce_line_ids + unbuild.consume_line_ids
+                ).stock_valuation_layer_ids.ids,
+            )
+        ]
+        self.assertEqual(
+            result["domain"],
+            domain,
+            "You should have as domain the ids of the stock valuation belonging "
+            "to the ids of the stock moves of produce_line and consume_line",
+        )
```

Comment:
you should not test the domain definition, but that this domain returns some layers, and that those layers are the same that you expect (search them by another mean).

Reply:
@ThiagoMForgeFlow Can you check @LoisRForgeFlow comment?

Reply:
It's done, thank you